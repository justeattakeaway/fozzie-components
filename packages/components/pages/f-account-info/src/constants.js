export const UPDATE_CONSUMER_DETAILS = 'updateConsumerDetails';
export const UPDATE_CONSUMER_DETAIL = 'updateConsumerDetail';

export const EVENT_SPINNER_STOP_LOADING = 'stop-spinner';

export const CONSUMER_DETAILS_URL = 'consumer';
export const CONSUMER_ADDRESSES_URL = 'consumer/addresses';

export const CONVERSATION_ID_NAME = 'x-je-conversation';
