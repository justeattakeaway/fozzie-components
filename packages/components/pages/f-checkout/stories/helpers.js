import {
    CHECKOUT_NOTE_TYPE_KITCHEN,
    CHECKOUT_NOTE_TYPE_COURIER
} from '../src/constants';

export const NOTE_TYPES = {
    courier: [CHECKOUT_NOTE_TYPE_COURIER],
    split: [CHECKOUT_NOTE_TYPE_COURIER, CHECKOUT_NOTE_TYPE_KITCHEN]
};

export const TENANTS = {
    uk: 'uk',
    au: 'au',
    nz: 'nz'
};

export const DELIVERY_TIMES = {
    later: 'later',
    unavailable: 'unavailable',
    now: 'now'
};

export const httpStatusCodes = {
    noResponse: 0,
    ok: 200,
    badRequest: 400,
    forbidden: 403,
    internalServerError: 500
};

export const httpMethods = {
    get: 'get',
    post: 'post',
    patch: 'patch'
};

export const ISSUES = {
    timeUnavailable: 'FULFILMENT_TIME_UNAVAILABLE',
    serviceTypeUnavailable: 'SERVICE_TYPE_UNAVAILABLE',
    restaurantNotTakingOrders: 'RESTAURANT_NOT_TAKING_ORDERS',
    geoLocationRequired: 'GEOLOCATION_REQUIRED',
    additionalItemsRequired: 'ADDITIONAL_ITEMS_REQUIRED',
    forbidden: 'forbidden'
};
