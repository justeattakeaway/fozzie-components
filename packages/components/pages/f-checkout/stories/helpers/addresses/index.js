import { auAddress, auAddresses } from './au';
import { itAddress, itAddresses } from './it';
import { nzAddress, nzAddresses } from './nz';
import { ukAddress, ukAddresses } from './uk';
import { esAddress, esAddresses } from './es';
import { ieAddress, ieAddresses } from './ie';

export const Address = {
    au: auAddress,
    ie: ieAddress,
    it: itAddress,
    nz: nzAddress,
    es: esAddress,
    uk: ukAddress
};

export const Addresses = {
    au: auAddresses,
    ie: ieAddresses,
    it: itAddresses,
    nz: nzAddresses,
    es: esAddresses,
    uk: ukAddresses
};
