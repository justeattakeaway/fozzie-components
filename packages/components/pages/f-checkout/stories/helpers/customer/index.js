import uk from './uk';
import au from './au';
import it from './it';
import nz from './nz';

export default {
    au,
    uk,
    it,
    nz
};
