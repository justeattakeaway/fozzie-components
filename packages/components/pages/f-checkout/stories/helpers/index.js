import {
    CHECKOUT_NOTE_TYPE_KITCHEN,
    CHECKOUT_NOTE_TYPE_COURIER
} from '../../src/constants';

export const NOTE_TYPES = {
    courier: [CHECKOUT_NOTE_TYPE_COURIER],
    split: [CHECKOUT_NOTE_TYPE_COURIER, CHECKOUT_NOTE_TYPE_KITCHEN]
};

export const TENANTS = {
    uk: 'uk',
    au: 'au',
    nz: 'nz'
};

export const DELIVERY_TIMES = {
    later: 'later',
    unavailable: 'unavailable-time',
    asap: 'asap'
};

export const httpStatusCodes = {
    noResponse: 0,
    ok: 200,
    badRequest: 400,
    forbidden: 403,
    internalServerError: 500
};

export const httpMethods = {
    get: 'get',
    post: 'post',
    patch: 'patch'
};

export const getSuccess = {
    method: httpMethods.get,
    responseStatus: httpStatusCodes.ok
};

export const getNoResponse = {
    method: httpMethods.get,
    responseStatus: httpStatusCodes.ok
};

export const postSuccess = {
    method: httpMethods.post,
    responseStatus: httpStatusCodes.ok
};

export const patchSuccess = {
    method: httpMethods.patch,
    responseStatus: httpStatusCodes.ok
};

export const authToken = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6ImpvZS5ibG9nZ3NAanVzdGVhdHRha2Vhd2F5LmNvbSIsImNyZWF0ZWRfZGF0ZSI6IjIwMjEtMDItMDhUMTA6Mjc6NDkuMTkzMDAwMFoiLCJuYW1lIjoiSm9lIEJsb2dncyIsImdsb2JhbF91c2VyX2lkIjoiVTdOUkFsV0FnNXpPZHNkUmdmN25rVHlvaTkwWEVvPSIsImdpdmVuX25hbWUiOiJKb2UiLCJmYW1pbHlfbmFtZSI6IkJsb2dncyIsImlhdCI6MTYxNTQ2OTUxNn0.VapH6uHnn4lHIkvN_mS9A9IVVWL0YPNE39gDDD-l7SU';

export const ISSUES = {
    timeUnavailable: 'FULFILMENT_TIME_UNAVAILABLE',
    serviceTypeUnavailable: 'SERVICE_TYPE_UNAVAILABLE',
    restaurantNotTakingOrders: 'RESTAURANT_NOT_TAKING_ORDERS',
    geoLocationRequired: 'GEOLOCATION_REQUIRED',
    additionalItemsRequired: 'ADDITIONAL_ITEMS_REQUIRED',
    forbidden: 'forbidden'
};

