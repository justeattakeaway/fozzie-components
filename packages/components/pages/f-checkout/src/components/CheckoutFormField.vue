<template>
    <form-field
        :value="customer[fieldName]"
        :name="translations.name"
        :input-type="inputType"
        :label-text="translations.label"
        :maxlength="maxLength"
        :has-error="hasError"
        :aria-invalid="hasError"
        :aria-describedby="hasError && translations.errorName"
        :aria-label="isPhoneNumber && formattedMobileNumberForScreenReader"
        @blur="hasInvalidErrorMessage && formFieldBlur()"
        @input="updateCustomerDetails({ [fieldName]: $event })">
        <template
            v-if="hasError"
            #error>
            <error-message
                :id="translations.errorName"
                data-js-error-message
                :data-test-id="translations[`${errorType}Error`]">
                {{ translations.validationMessages[errorType] }}
            </error-message>
        </template>
    </form-field>
</template>

<script>
import ErrorMessage from '@justeat/f-error-message';
import '@justeat/f-error-message/dist/f-error-message.css';
import FormField from '@justeat/f-form-field';
import '@justeat/f-form-field/dist/f-form-field.css';
import { mapState, mapActions } from 'vuex';
import { VALIDATIONS, VUEX_CHECKOUT_MODULE } from '../constants';

export default {
    components: { FormField, ErrorMessage },

    props: {
        fieldName: {
            type: String,
            required: true
        },

        maxLength: {
            type: String,
            default: '100'
        }
    },

    /**
     * Provide/Inject allows component to inherit `Checkout`
     * validator scope, `$v`.
    */
    inject: ['$v'],

    computed: {
        ...mapState(VUEX_CHECKOUT_MODULE, [
            'customer'
        ]),

        translations () {
            return {
                ...this.$t(`formFields.customer.${this.fieldName}`),
                name: `${this.kebabCase}`,
                errorName: `${this.kebabCase}-error`,
                emptyError: `error-${this.kebabCase}-empty`,
                invalidError: `error-${this.kebabCase}-invalid`
            };
        },

        validations () {
            return this.$v[VALIDATIONS.customer][this.fieldName];
        },

        errorType () {
            if (this.hasError) {
                return this.hasRequiredError ? 'required' : 'invalid';
            }
            return null;
        },

        inputType () {
            const inputTypes = {
                mobileNumber: 'tel',
                email: 'email',
                default: 'text'
            };

            return inputTypes[this.fieldName] || inputTypes.default;
        },

        isPhoneNumber () {
            return this.inputType === 'tel';
        },

        hasValidationMessages () {
            return !!this.translations.validationMessages;
        },

        hasInvalidErrorMessage () {
            return this.hasValidationMessages && !!this.translations.validationMessages.invalid;
        },

        isEmpty () {
            return this.validations.$dirty;
        },

        hasRequiredError () {
            return this.isEmpty && !this.validations.required;
        },

        hasInvalidError () {
            return !!this.hasInvalidErrorMessage && this.isEmpty && !this.validations[this.fieldName];
        },

        hasError () {
            return this.hasValidationMessages && (this.hasRequiredError || this.hasInvalidError);
        },

        kebabCase () {
            return this.fieldName.replace(/([a-z0-9]|(?=[A-Z]))([A-Z])/g, '$1-$2').toLowerCase();
        },

        formattedMobileNumberForScreenReader () {
            return this.customer.mobileNumber ? [...this.customer.mobileNumber].join(' ') : '';
        }
    },

    methods: {
        ...mapActions(VUEX_CHECKOUT_MODULE, [
            'updateCustomerDetails'
        ]),

        formFieldBlur () {
            if (this.validations) {
                this.validations.$touch();
            }
        }
    }
};
</script>
