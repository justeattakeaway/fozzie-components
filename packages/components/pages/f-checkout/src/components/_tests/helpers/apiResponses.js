import getCheckoutResponse from '../../../../stories/api/responses/getCheckout/getCheckout';
import getBasketResponse from '../../../../stories/api/responses/getBasket/getBasket';
import { getAvailableTimes } from '../../../../stories/api/responses/getAvailableTimes';
import geoLocationDetailsResponse from '../../../../stories/api/responses/getGeoLocation';
import { getNotesConfig } from '../../../../stories/api/responses/getNotes';
import { CHECKOUT_METHOD_DELIVERY } from '../../../constants';

// getGeoLocation
export const geoLocationDetails = geoLocationDetailsResponse[0].payload;

// getCheckout
export const checkoutDelivery = getCheckoutResponse();

// getBasket
export const basketDelivery = getBasketResponse();
export const basketInvalidProducts = getBasketResponse(CHECKOUT_METHOD_DELIVERY, 'invalidProduct');
export const basketOfflineProducts = getBasketResponse(CHECKOUT_METHOD_DELIVERY, 'offlineProduct');
export const basketDeliveryAgeRestricted = getBasketResponse(CHECKOUT_METHOD_DELIVERY, 'ageRestriction');

// getAvailableTimes
export const checkoutAvailableFulfilment = getAvailableTimes();

// getSplitNotes
export const splitNotesConfig = getNotesConfig(true);
