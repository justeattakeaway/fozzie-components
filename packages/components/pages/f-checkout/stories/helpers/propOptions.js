import { CHECKOUT_METHOD_COLLECTION, CHECKOUT_METHOD_DELIVERY, CHECKOUT_METHOD_DINEIN } from '../../src/constants';

const serverTimeout = 'Server timeout';
const serverTimeoutIssue = 'timeout';
const accessForbiddenErrorCode = '403';
export const timeUnavailable = 'time-unavailable';

const patchCheckoutErrorOptions = {
    None: null,
    'Restaurant Not Taking Orders Issue (Response from server but order not fulfillable)': 'restaurant-not-taking-orders',
    'ServiceType is not available (Response from server but order not fulfillable)': 'service-type-unavailable',
    'Additional Items Required Issue (Response from server but order not fulfillable)': 'additional-items-required',
    'Checkout Error (Response from server is an error)': 'SERVER',
    'Access Forbidden (Response from server is 403)': accessForbiddenErrorCode,
    'Selected time no longer available': timeUnavailable,
    'Geolocation required': 'geolocation-required',
    [serverTimeout]: serverTimeoutIssue
};

const createGuestErrorOptions = {
    None: null,
    'Create Guest Error': 'error'
};

const getCheckoutOptions = {
    None: null,
    'Error - Access Forbidden (Response from server is an error)': accessForbiddenErrorCode,
    'Error - Any other (Response from server is an error)': '500',
    'Scheduled Time - Asap': 'user-selected-asap',
    'Scheduled Time - Later': 'user-selected-later',
    'Scheduled Time - Unavailable': `user-selected-${timeUnavailable}`,
    [serverTimeout]: serverTimeoutIssue
};

const getBasketOptions = {
    None: null,
    'Error - Basket contains invalid products': 'invalid-products',
    'Error - Basket contains offline products': 'offline-products',
    'Restriction - Age restricted': 'age-restriction'
};

const placeOrderErrorOptions = {
    None: null,
    'Place Order Duplicate Error (Response from server is an error)': 'duplicate',
    [serverTimeout]: serverTimeoutIssue
};

const fulfilmentTimeErrors = {
    none: null,
    'Error - No Time Available': timeUnavailable,
    'Available Time Issues': 'issues'
};

const noteTypeOptions = {
    'Legacy notes': null,
    'Combined note': 'get-notes-config',
    'Delivery and Kitchen notes': 'get-notes-config-split'
};

const serviceTypeOptions = [CHECKOUT_METHOD_COLLECTION, CHECKOUT_METHOD_DELIVERY, CHECKOUT_METHOD_DINEIN, 'invalid-url'];

export const propOptions = {
    noteTypeOptions,
    patchCheckoutErrorOptions,
    createGuestErrorOptions,
    getCheckoutOptions,
    getBasketOptions,
    placeOrderErrorOptions,
    fulfilmentTimeErrors,
    serviceTypeOptions,
    timeUnavailable
};
