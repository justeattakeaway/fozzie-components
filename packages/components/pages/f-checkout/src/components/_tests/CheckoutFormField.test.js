import Vuex from 'vuex';
import { VueI18n } from '@justeat/f-globalisation';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import CheckoutFormField from '../CheckoutFormField.vue';
import { i18n, createStore } from './helpers/setup';
import { VALIDATIONS } from '../../constants';

const localVue = createLocalVue();

localVue.use(VueI18n);
localVue.use(Vuex);

const $v = {
    [VALIDATIONS.customer]: {
        mobileNumber: {
            $dirty: false,
            mobileNumber: false
        },
        firstName: {
            $dirty: false
        },
        lastName: {
            $dirty: false
        },
        email: {
            $dirty: false,
            email: false,
            $touch: jest.fn()
        }
    },
    [VALIDATIONS.address]: {
        locality: {
            $dirty: false,
            required: true
        },
        line1: {
            $dirty: false,
            required: false
        },
        city: {
            $dirty: false,
            required: false
        },
        administrativeArea: {
            $dirty: false,
            required: true
        },
        postcode: {
            $dirty: false,
            required: true,
            isValidPostcode: false
        }
    },
    [VALIDATIONS.dineIn]: {
        tableIdentifier: {
            $dirty: false
        }
    }
};

describe('CheckoutFormField', () => {
    const propsData = {
        fieldName: 'firstName',
        fieldType: 'customer'
    };

    it('should be defined', () => {
        // Arrange
        const wrapper = shallowMount(CheckoutFormField, {
            i18n,
            store: createStore(),
            localVue,
            propsData,
            provide: () => ({
                $v
            })
        });

        // Assert
        expect(wrapper.exists()).toBe(true);
    });

    describe('computed ::', () => {
        describe('translations ::', () => {
            it('should return translations from with id names', () => {
                // Arrange & Act
                const wrapper = shallowMount(CheckoutFormField, {
                    i18n,
                    store: createStore(),
                    localVue,
                    propsData,
                    provide: () => ({
                        $v
                    })
                });

                // Assert
                expect(wrapper.vm.translations).toMatchSnapshot();
            });
        });

        describe('errorType ::', () => {
            let hasErrorSpy;
            let hasRequiredErrorSpy;

            beforeEach(() => {
                hasErrorSpy = jest.spyOn(CheckoutFormField.computed, 'hasError');
                hasRequiredErrorSpy = jest.spyOn(CheckoutFormField.computed, 'hasRequiredError');
            });

            afterEach(() => {
                jest.clearAllMocks();
            });

            it.each([
                [false, true, null],
                [false, false, null],
                [true, false, 'invalid'],
                [true, true, 'required']
            ])('when `hasError` is %s and `hasRequiredError` is %s, should return %s', (hasError, hasRequiredError, expected) => {
                // Arrange
                hasErrorSpy.mockReturnValue(hasError);
                hasRequiredErrorSpy.mockReturnValue(hasRequiredError);

                // Act
                const wrapper = shallowMount(CheckoutFormField, {
                    i18n,
                    store: createStore(),
                    localVue,
                    propsData,
                    provide: () => ({
                        $v
                    })
                });

                // Assert
                expect(wrapper.vm.errorType).toEqual(expected);
            });
        });

        describe('hasInvalidError ::', () => {
            let hasInvalidErrorMessageSpy;
            let isEmptySpy;
            let validationsSpy;

            beforeEach(() => {
                hasInvalidErrorMessageSpy = jest.spyOn(CheckoutFormField.computed, 'hasInvalidErrorMessage');
                isEmptySpy = jest.spyOn(CheckoutFormField.computed, 'isEmpty');
                validationsSpy = jest.spyOn(CheckoutFormField.computed, 'validations');
            });

            afterEach(() => {
                jest.clearAllMocks();
            });

            describe('when validations for field invalid error is true ::', () => {
                beforeEach(() => {
                    validationsSpy.mockReturnValue({
                        email: {
                            email: true
                        }
                    });
                });

                it.each([
                    [true, true, false],
                    [true, false, false],
                    [false, true, false],
                    [false, false, false]
                ])('when `hasInvalidErrorMessage` is %s and `isEmpty` is %s should return %s', (hasInvalidErrorMessage, isEmpty, expected) => {
                    // Arrange
                    hasInvalidErrorMessageSpy.mockReturnValue(hasInvalidErrorMessage);
                    isEmptySpy.mockReturnValue(isEmpty);

                    // Act
                    const wrapper = shallowMount(CheckoutFormField, {
                        i18n,
                        store: createStore(),
                        localVue,
                        propsData: {
                            ...propsData,
                            fieldName: 'email'
                        },
                        provide: () => ({
                            $v
                        })
                    });

                    // Assert
                    expect(wrapper.vm.hasInvalidError).toEqual(expected);
                });
            });

            describe('when validations for field invalid error is false ::', () => {
                beforeEach(() => {
                    validationsSpy.mockReturnValue({
                        email: {
                            email: false
                        }
                    });
                });

                it('should return false', () => {
                    // Act
                    const wrapper = shallowMount(CheckoutFormField, {
                        i18n,
                        store: createStore(),
                        localVue,
                        propsData: {
                            ...propsData,
                            fieldName: 'email'
                        },
                        provide: () => ({
                            $v
                        })
                    });

                    // Assert
                    expect(wrapper.vm.hasInvalidError).toEqual(false);
                });
            });
        });

        describe('hasError ::', () => {
            let hasValidationMessagesSpy;

            beforeEach(() => {
                hasValidationMessagesSpy = jest.spyOn(CheckoutFormField.computed, 'hasValidationMessages');
            });

            afterEach(() => {
                jest.restoreAllMocks();
            });

            describe('when `hasValidationMessages` is true', () => {
                let wrapper;

                beforeEach(() => {
                    hasValidationMessagesSpy.mockReturnValue(true);
                });

                afterEach(() => {
                    jest.clearAllMocks();
                });

                it.each([
                    [true, true, true],
                    [true, false, true],
                    [true, true, false],
                    [false, false, false]
                ])('should return %s when `hasRequiredError` is %s and `hasInvalidError` is %s', (expected, hasRequiredError, hasInvalidError) => {
                    // Arrange & Act
                    jest.spyOn(CheckoutFormField.computed, 'hasRequiredError').mockReturnValue(hasRequiredError);
                    jest.spyOn(CheckoutFormField.computed, 'hasInvalidError').mockReturnValue(hasInvalidError);

                    wrapper = shallowMount(CheckoutFormField, {
                        i18n,
                        store: createStore(),
                        localVue,
                        propsData,
                        provide: () => ({
                            $v
                        })
                    });

                    // Assert
                    expect(wrapper.vm.hasError).toEqual(expected);
                });
            });

            describe('when `hasValidationMessages` is false', () => {
                afterEach(() => {
                    jest.clearAllMocks();
                });

                it('should return false', () => {
                    // Arrange & Act
                    hasValidationMessagesSpy.mockReturnValue(false);

                    const wrapper = shallowMount(CheckoutFormField, {
                        i18n,
                        store: createStore(),
                        localVue,
                        propsData,
                        provide: () => ({
                            $v
                        })
                    });

                    // Assert
                    expect(wrapper.vm.hasError).toEqual(false);
                });
            });
        });

        describe('hasValidationMessages ::', () => {
            const translations = {};
            const validationMessages = {
                invalid: 'Invalid message'
            };

            afterEach(() => {
                jest.clearAllMocks();
            });

            it('should return false if `translations` do not have `validationMessages', () => {
                // Arrange & Act
                jest.spyOn(CheckoutFormField.computed, 'translations').mockReturnValue(translations);

                const wrapper = shallowMount(CheckoutFormField, {
                    i18n,
                    store: createStore(),
                    localVue,
                    propsData,
                    provide: () => ({
                        $v
                    })
                });

                // Assert
                expect(wrapper.vm.hasValidationMessages).toEqual(false);
            });

            it('should return false if `translations` do not have `validationMessages', () => {
                // Arrange & Act
                translations.validationMessages = validationMessages;
                jest.spyOn(CheckoutFormField.computed, 'translations').mockReturnValue(translations);

                const wrapper = shallowMount(CheckoutFormField, {
                    i18n,
                    store: createStore(),
                    localVue,
                    propsData,
                    provide: () => ({
                        $v
                    })
                });

                // Assert
                expect(wrapper.vm.hasValidationMessages).toEqual(true);
            });
        });

        describe('kebabCase ::', () => {
            it.each([
                ['firstName', 'customer', 'first-name'],
                ['lastName', 'customer', 'last-name'],
                ['email', 'customer', 'email'],
                ['line1', 'address', 'line-1'],
                ['city', 'address', 'city'],
                ['tableIdentifier', 'dineIn', 'table-identifier']
            ])('should convert `fieldName` value to kebab case', (fieldName, fieldType, expected) => {
                // Arrange & Act
                const wrapper = shallowMount(CheckoutFormField, {
                    i18n,
                    store: createStore(),
                    localVue,
                    propsData: {
                        fieldName,
                        fieldType
                    },
                    provide: () => ({
                        $v
                    })
                });

                // Assert
                expect(wrapper.vm.kebabCase).toEqual(expected);
            });
        });

        describe('formattedMobileNumberForScreenReader ::', () => {
            it('should return the mobile number with spaces after every character', () => {
                const expectedMobileNumber = '+ 4 4 7 1 1 1 1 1 1 1 1 1';

                // Act
                const wrapper = shallowMount(CheckoutFormField, {
                    store: createStore(),
                    i18n,
                    localVue,
                    propsData: {
                        fieldName: 'mobileNumber',
                        fieldType: 'customer'
                    },
                    provide: () => ({
                        $v
                    })
                });

                // Assert
                expect(wrapper.vm.formattedMobileNumberForScreenReader).toEqual(expectedMobileNumber);
            });
        });

        describe('groupedProps ::', () => {
            it('should return props for grouped form fields', () => {
                // Arrange & Act
                const wrapper = shallowMount(CheckoutFormField, {
                    i18n,
                    store: createStore(),
                    localVue,
                    propsData,
                    provide: () => ({
                        $v
                    })
                });

                // Assert
                expect(wrapper.vm.groupedProps).toMatchSnapshot();
            });
        });
    });

    describe('methods ::', () => {
        describe('formFieldBlur ::', () => {
            afterEach(() => {
                jest.clearAllMocks();
            });

            it('should call `touch` for validation', () => {
                // Arrange
                const fieldName = 'email';
                const touchSpy = jest.spyOn($v[VALIDATIONS.customer].email, '$touch');

                const wrapper = shallowMount(CheckoutFormField, {
                    i18n,
                    store: createStore(),
                    localVue,
                    propsData: {
                        ...propsData,
                        fieldName
                    },
                    provide: () => ({
                        $v
                    })
                });

                // Act
                wrapper.vm.formFieldBlur(fieldName);

                // Assert
                expect(touchSpy).toHaveBeenCalled();
            });
        });
    });
});
