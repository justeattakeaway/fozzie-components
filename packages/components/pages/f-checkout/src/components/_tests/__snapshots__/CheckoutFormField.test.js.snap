// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutFormField computed :: translations :: should return translations from with id names 1`] = `
Object {
  "emptyError": "error-first-name-empty",
  "errorName": "first-name-error",
  "invalidError": "error-first-name-invalid",
  "label": "First name",
  "name": "first-name",
  "validationMessages": Object {
    "required": "Please enter your first name",
  },
}
`;
